cpp/oneapi/dal/algo/kmeans/test/multi_proc_fixture.hpp

```diff
@@ -0,0 +1,669 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <limits>
+#include <cmath>
+
+#include "oneapi/dal/algo/kmeans/train.hpp"
+#include "oneapi/dal/algo/kmeans/infer.hpp"
+#include "oneapi/dal/algo/kmeans/test/data.hpp"
+
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+#include "oneapi/dal/test/engine/metrics/clustering.hpp"
+#include "oneapi/dal/spmd/communicator.hpp"
+
+namespace oneapi::dal::kmeans::test {
+
+namespace te = dal::test::engine;
+namespace la = dal::test::engine::linalg;
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+using kmeans_types = COMBINE_TYPES((float, double), (kmeans::method::lloyd_dense));
+
+template <typename Backend, typename TestType, typename Derived>
+class kmeans_mpi_test : public te::crtp_par_algo_fixture<TestType, Derived> {
```

Comment:
kmeans_mpi_test -> rename to kmeans_spmd_backend_test

Reply:
Replaced mpi/multi_proc with spmd_backend